Moves secure closet locking to Right click

SgtHunk opened the pull request:
## About The Pull Request

Adds Right Click functionality (taken from https://github.com/tgstation/tgstation/pull/58469 which sadly got stale'd)
Moves the secure closet's Alt click function (which toggles the lock of the secure closet) to Right click. This does NOT remove LClick's functionality of unlocking a secure closet.

## Why It's Good For The Game

Less alt+click interaction, more use of the now-available right click.

## Changelog
:cl:
qol: You can now unlock/lock secure lockers with Right click rather than using Alt+click.
/:cl:

TJatPBnJ committed:
adds right click signal

SgtHunk committed:
Merge remote-tracking branch 'upstream/master'

SgtHunk committed:
Merge branch 'master' into upstream/master

SgtHunk committed:
Merge remote-tracking branch 'upstream/master'

SgtHunk committed:
Merge remote-tracking branch 'upstream/master'

SgtHunk committed:
Makes lockers lock/unlock on RClick rather than Altclick

Mothblocks commented on code/game/objects/structures/crates_lockers/closets.dm:
```diff
@@ -448,14 +448,11 @@
-		return
-	else
-		togglelock(user)
+		return SECONDARY_ATTACK_CONTINUE_CHAIN
```
I don't think it makes much sense to continue `afterattack` when nothing has happened. This entire thing should probably just check if it *can* lock, lock it, then always return `SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN`.

Mothblocks reviewed (changes requested):
Good PR :+1:

SgtHunk committed:
changes it so it follows a better logic

Mothblocks reviewed (changes requested):
So something I just noticed now, you're regressing behavior--alt-click works with full hands and this only works with a normal hand.

Change this to raw `RightClick` (I'm assuming we have that anyway), and perform the same checks. I think this behavior is important since you've usually just taken something out of a locker.

SgtHunk commented:
It seems like there's no RightClick proc by itself, although attackby_secondary and attack_hand_secondary exist

Mothblocks commented:
We can fix that later, but for now just connect to Click and check params for `RIGHT_CLICK` the same way other stuff does.

SgtHunk commented:
Going to close this until I both: have the time for it and figure it out, will probably return to it if no one else also PRs this.

SgtHunk committed:
Makes the locker use right click

Mothblocks commented on code/_onclick/click.dm:
```diff
@@ -77,8 +77,10 @@
-
 	var/list/modifiers = params2list(params)
+	if(LAZYACCESS(modifiers, RIGHT_CLICK))
+		if(RightClickOn(A))
```
I'm very wary of this, and probably need to do some testing of my own.

Are you CONFIDENT this preserves the behavior there is now, where if I right-click something it'll act like a left-click if nothing hooks to this? As in--right clicking items still picks them up, right-clicking a machine will still use it, etc.

SgtHunk replied:
Good catch, you're not wrong. It did end up removing RClick's passive functionality of being LClick.

Mothblocks commented on code/_onclick/click.dm:
```diff
@@ -300,6 +302,17 @@
+	return target.RightClick(src)
+
+/atom/proc/RightClick(mob/user)
+	if(SEND_SIGNAL(src, COMSIG_CLICK_RIGHT, user) & COMPONENT_CANCEL_CLICK_ALT)
```
Make a new flag instead of using `COMPONENT_CANCEL_CLICK_ALT`.

Mothblocks commented on code/game/objects/structures/crates_lockers/closets.dm:
```diff
@@ -448,14 +448,10 @@
+/obj/structure/closet/RightClick(mob/user, modifiers)
+	if(!opened && secure)
 		togglelock(user)
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
```
This isn't a valid return type for this, it expects a boolean.

SgtHunk committed:
Adresses all issues

Makes cancel right click flag, fixes interactions breaking, returns TRUE

Mothblocks commented on code/_onclick/click.dm:
```diff
@@ -77,8 +77,10 @@
-
 	var/list/modifiers = params2list(params)
+	if(LAZYACCESS(modifiers, RIGHT_CLICK))
+		if(RightClickOn(A))
```
You don't appear to pass params.

Mothblocks commented on code/_onclick/click.dm:
```diff
@@ -300,6 +302,16 @@
+/mob/proc/RightClickOn(atom/target, params)
+	return target.RightClick(src)
+
+/atom/proc/RightClick(mob/user)
```
This and `RightClickOn` need to be documented, properly. Not just "right click", but what is params? What does the return value do?

https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md

Mothblocks reviewed (changes requested):
We're really close, I promise.

SgtHunk committed:
Clean-up and documentation

Params was unnecesary haha
Also documents it to my best knowledge
Wouldn't be surprised if I missed something

Mothblocks commented on code/_onclick/click.dm:
```diff
@@ -304,10 +304,23 @@
  * Right click
+ *
+ * Used for Rightclick interactions, in similar fashion of AltClick.
+ * Returns RightClick().
```
This is still not good documentation, it's just documenting an implementation detail.

What does the return value MEAN?

Mothblocks commented on code/_onclick/click.dm:
```diff
@@ -304,10 +304,23 @@
 /**
  * Right click
+ *
+ * Used for Rightclick interactions, in similar fashion of AltClick.
```
right-click (change it in the other place too)

Mothblocks commented on code/_onclick/click.dm:
```diff
@@ -304,10 +304,23 @@
+ * Proc used for right-clicking
+ *
+ * Used for Rightclick interactions, called by RightClickOn().
+ * Returns TRUE upon checking if a right-click has happened.
```
This doesn't look accurate, it's only if the signal is hooked.

SgtHunk committed:
i probably missed something again

Mothblocks commented on code/_onclick/click.dm:
```diff
@@ -300,6 +302,29 @@
+ * Proc used for right-clicking
+ *
+ * Used for right-click interactions, called by [mob/proc/RightClickOn].
+ * Returns TRUE if a signal hooked [COMSIG_CLICK_RIGHT] right-click has happened.
```
Again, it's not JUST when a signal hooks--it returns TRUE if the click should not continue the click chain, for whatever reason.

Mothblocks commented on code/_onclick/click.dm:
```diff
@@ -300,6 +302,29 @@
+ * Right click
+ *
+ * Used for right-clicking interactions, in similar fashion of AltClick.
+ * Returns [atom/proc/RightClick] on the atom being right-clicked. This sends a signal.
```
Again, what does this return value *mean*? I shouldn't have to check other documentation to understand this one. It should have the same description of "returns TRUE if the click chain should not continue", or similar.

Sorry if this all seems like nit-picking, but accurate and useful documentation is super important.

SgtHunk committed:
i almost forgot about this pr

and im still confused about specifics relating documentation god i'm dumb

Mothblocks commented:
Looks good now!
